activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -87,8 +87,12 @@ def indexes(table_name) # :nodoc:
           scope = quoted_scope(table_name)
 
           result = query(<<~SQL, "SCHEMA")
-            SELECT distinct i.relname, d.indisunique, d.indkey, pg_get_indexdef(d.indexrelid), t.oid,
-                            pg_catalog.obj_description(i.oid, 'pg_class') AS comment, d.indisvalid
+            SELECT distinct i.relname, d.indisunique, d.indkey, pg_get_indexdef(d.indexrelid),
+                            pg_catalog.obj_description(i.oid, 'pg_class') AS comment, d.indisvalid,
+                            ARRAY(
+                              SELECT pg_get_indexdef(d.indexrelid, k + 1, true)
+                              FROM generate_subscripts(d.indkey, 1) AS k
```

Comment:
I'm 30% sure this needs an `ORDER BY k` to be Technically Correct 🤔 

This also feels like a behaviour change for non-column-reference expressions. Is that true?

Reply:
I think the `ORDER BY k` is not needed. `indkey` returns a vector (in postgres terminology) with column numbers from the table. So for table `users(id, name, email, created_at)` and index on `(created_at, email)` it returns `[4, 3]`.

> This also feels like a behaviour change for non-column-reference expressions.

Can you provide an example?

Reply:
My concern was for an index that mixed columns and non-column expressions, but per Discord conversation, I now see that the changed codepath (around line 125 below) is only taken for indexes that consist exclusively of table columns: as soon as any expression is present, we hit `if indkey.include?(0)`, and things diverge.

(It separately seems less-than-ideal that we skip over the "richer" behaviour for those index columns that _are_ non-expression references to table columns, but that's not relevant to this change.)

Reply:
(I'm also happy to assume that `generate_subscripts` produces rows in a defined order within the immediate subquery, even without an explicit `ORDER BY`, just on the basis that set returning functions are weird.)

Reply:
Added `ORDER BY k` to be sure. I see that was used in other examples on the internets, but I did not get it why we need to use it, because seems like `generate_subscripts` result should already be sorted.

> (It separately seems less-than-ideal that we skip over the "richer" behaviour for those index columns that are non-expression references to table columns, but that's not relevant to this change.)

Probably I should got to sleep, but I did not get what is the problem here? 🤔 I would appreciate if you can provide a concrete example where the new approach won't work.

Reply:
@matthewd Can you please reconsider this PR?

Reply:
Rebased.